datumaro/plugins/open_images_format.py

```diff
@@ -137,11 +142,16 @@ def set_parents_from_node(node, category):
         set_parents_from_node(root_node, root_category)
 
     def _load_items(self):
+        images_dir = osp.join(self._dataset_dir, OpenImagesPath.IMAGES_DIR)
+
         image_paths_by_id = {
-            osp.splitext(osp.basename(path))[0]: path
-            for path in find_images(
-                osp.join(self._dataset_dir, 'images'),
-                recursive=True, max_depth=1)
+            # the first component of `path_parts` is the subset name
+            '/'.join(path_parts[1:]): path
+            for path in find_images(images_dir, recursive=True)
+            for path_parts in [split_path(
+                osp.splitext(osp.relpath(path, images_dir))[0],
+            )]
+            if len(path_parts) > 1
```

Comment:
```suggestion
            if 1 < len(path_parts)
```
Please, use numbers in comparisons in ascending order in this project.